scheduler/generic_sched_test.go

```diff
@@ -3644,24 +3644,94 @@ func TestServiceSched_StopAfterClientDisconnect(t *testing.T) {
 	ci.Parallel(t)
 
 	cases := []struct {
-		stop        time.Duration
+		jobSpec     func() *structs.Job
```

Comment:
```suggestion
		jobSpecFn     func(*structs.Job)
```

A small change if you would like to avoid having to call `mock.Job()` inside each test case. The main test loop could call it once and feed it to each case for modifications.